core/network/helpers/message_read_writer.hpp

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_MESSAGE_READ_WRITER_HPP
+#define KAGOME_MESSAGE_READ_WRITER_HPP
+
+#include <functional>
+#include <memory>
+#include <vector>
+#include <gsl/span>
+#include <boost/assert.hpp>
+#include <boost/system/error_code.hpp>
+
+#include "outcome/outcome.hpp"
+
+namespace kagome::network {
+  struct NoSink {};
```

Comment:
Please comment, it's not that hard to guess what it is, but still it would be a bit simpler to read if commented.